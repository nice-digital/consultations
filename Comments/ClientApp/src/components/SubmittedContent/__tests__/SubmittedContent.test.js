import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import { SubmittedContent } from "../SubmittedContent";

const fakeProps = {
	organisationName: "My Org",
	isOrganisationCommenter: false,
	isLead: false,
	consultationState:{
		supportsDownload: true,
		leadHasBeenSentResponse: false,
	},
	consultationId: 1,
	basename: "something",
	isSubmitted: true,
	linkToReviewPage: false,
};

Date.now = jest.fn(() => new Date(Date.UTC(2022, 7, 9, 8)).valueOf());

test("should match snapshot for individual commenters when displayed on the review page", () => {
	const {container} = render(<SubmittedContent {...fakeProps} />);
	expect(container).toMatchSnapshot();
});

test("should match snapshot for individual commenters when displayed on the submitted page", () => {
	let localProps = {...fakeProps};
	localProps.linkToReviewPage = true;
	const {container} = render(<SubmittedContent {...localProps} />, {wrapper: MemoryRouter});
	expect(container).toMatchSnapshot();
});

test("should match snapshot for organisation commenter when displayed on the review page", () => {
	let localProps = {...fakeProps};
	localProps.isOrganisationCommenter = true;
	const {container} = render(<SubmittedContent {...localProps} />);
	expect(container).toMatchSnapshot();
});

test("should match snapshot for organisation commenter when displayed on the submitted page", () => {
	const localProps = {...fakeProps};
	localProps.isOrganisationCommenter = true;
	localProps.linkToReviewPage = true;
	const {container} = render(<SubmittedContent {...localProps} />, {wrapper: MemoryRouter});
	expect(container).toMatchSnapshot();
});

test("should match snapshot for organisation lead when displayed on the review page without any responses from their org", () => {
	const localProps = {...fakeProps};
	localProps.isLead = true;
	const {container} = render(<SubmittedContent {...localProps} />, {wrapper: MemoryRouter});
	expect(container).toMatchSnapshot();
});

test("should match snapshot for organisation lead when displayed on the review page with responses from their org", () => {
	const localProps = {...fakeProps};
	localProps.isLead = true;
	localProps.consultationState.leadHasBeenSentResponse = true;
	const {container} = render(<SubmittedContent {...localProps} />, {wrapper: MemoryRouter});
	expect(container).toMatchSnapshot();
});

test("should match snapshot for organisation lead when displayed on the submitted page", () => {
	const localProps = {...fakeProps};
	localProps.isLead = true;
	localProps.linkToReviewPage = true;
	localProps.consultationState.leadHasBeenSentResponse = true;
	const {container} = render(<SubmittedContent {...localProps} />, {wrapper: MemoryRouter});
	expect(container).toMatchSnapshot();
});
