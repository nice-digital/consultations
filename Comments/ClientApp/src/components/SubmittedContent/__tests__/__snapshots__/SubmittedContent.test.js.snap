// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ClientApp]  Submitted page should match snapshot for individual commenters when displayed on the review page 1`] = `
<Fragment>
  <h2>
    Your response was submitted 
     on 
    <n
      ago={false}
      calendar={false}
      decimal={false}
      element={null}
      filter={[Function]}
      format="D MMMM YYYY"
      fromNow={false}
      interval={60000}
      local={false}
      onChange={[Function]}
      titleFormat=""
      toNow={false}
      trim={false}
      unit={null}
      unix={false}
      utc={false}
      withTitle={false}
    />
    .
  </h2>
  <ul
    className="list list--unstyled"
  >
    <li>
      <a
        href="something/api/exportexternal/1"
        onClick={[Function]}
      >
        Download submitted response (Excel)
      </a>
    </li>
  </ul>
  <h2>
    What happens next?
  </h2>
  <p>
    We will review all the submissions received for this consultation. Our response will be published on the website around the time the guidance is published.
  </p>
  <hr />
</Fragment>
`;

exports[`[ClientApp]  Submitted page should match snapshot for individual commenters when displayed on the submitted page 1`] = `
<Fragment>
  <h2>
    Your response was submitted 
     on 
    <n
      ago={false}
      calendar={false}
      decimal={false}
      element={null}
      filter={[Function]}
      format="D MMMM YYYY"
      fromNow={false}
      interval={60000}
      local={false}
      onChange={[Function]}
      titleFormat=""
      toNow={false}
      trim={false}
      unit={null}
      unix={false}
      utc={false}
      withTitle={false}
    />
    .
  </h2>
  <ul
    className="list list--unstyled"
  >
    <li>
      <Link
        data-qa-sel="review-submitted-comments"
        to="/1/review"
      >
        Review your response
      </Link>
    </li>
    <li>
      <a
        href="something/api/exportexternal/1"
        onClick={[Function]}
      >
        Download submitted response (Excel)
      </a>
    </li>
  </ul>
  <h2>
    What happens next?
  </h2>
  <p>
    We will review all the submissions received for this consultation. Our response will be published on the website around the time the guidance is published.
  </p>
  <hr />
</Fragment>
`;

exports[`[ClientApp]  Submitted page should match snapshot for organisation commenter when displayed on the review page 1`] = `
<Fragment>
  <h2>
    Your response was submitted 
    to My Org
     on 
    <n
      ago={false}
      calendar={false}
      decimal={false}
      element={null}
      filter={[Function]}
      format="D MMMM YYYY"
      fromNow={false}
      interval={60000}
      local={false}
      onChange={[Function]}
      titleFormat=""
      toNow={false}
      trim={false}
      unit={null}
      unix={false}
      utc={false}
      withTitle={false}
    />
    .
  </h2>
  <ul
    className="list list--unstyled"
  >
    <li>
      <a
        href="something/api/exportexternal/1"
        onClick={[Function]}
      >
        Download submitted response (Excel)
      </a>
    </li>
  </ul>
  <h2>
    What happens next?
  </h2>
  <p>
    My Org
     will review all the submissions received for this consultation.
  </p>
  <p>
    NICE's response to all the submissions received will be published on the website around the time the final guidance is published.
  </p>
  <hr />
</Fragment>
`;

exports[`[ClientApp]  Submitted page should match snapshot for organisation commenter when displayed on the submitted page 1`] = `
<Fragment>
  <h2>
    Your response was submitted 
    to My Org
     on 
    <n
      ago={false}
      calendar={false}
      decimal={false}
      element={null}
      filter={[Function]}
      format="D MMMM YYYY"
      fromNow={false}
      interval={60000}
      local={false}
      onChange={[Function]}
      titleFormat=""
      toNow={false}
      trim={false}
      unit={null}
      unix={false}
      utc={false}
      withTitle={false}
    />
    .
  </h2>
  <ul
    className="list list--unstyled"
  >
    <li>
      <Link
        data-qa-sel="review-submitted-comments"
        to="/1/review"
      >
        Review your response
      </Link>
    </li>
    <li>
      <a
        href="something/api/exportexternal/1"
        onClick={[Function]}
      >
        Download submitted response (Excel)
      </a>
    </li>
  </ul>
  <h2>
    What happens next?
  </h2>
  <p>
    My Org
     will review all the submissions received for this consultation.
  </p>
  <p>
    NICE's response to all the submissions received will be published on the website around the time the final guidance is published.
  </p>
  <hr />
</Fragment>
`;

exports[`[ClientApp]  Submitted page should match snapshot for organisation lead when displayed on the review page with responses from their org 1`] = `
<Fragment>
  <h2>
    Your response was submitted 
     on 
    <n
      ago={false}
      calendar={false}
      decimal={false}
      element={null}
      filter={[Function]}
      format="D MMMM YYYY"
      fromNow={false}
      interval={60000}
      local={false}
      onChange={[Function]}
      titleFormat=""
      toNow={false}
      trim={false}
      unit={null}
      unix={false}
      utc={false}
      withTitle={false}
    />
    .
  </h2>
  <ul
    className="list list--unstyled"
  >
    <li>
      <a
        href="something/api/exportexternal/1"
        onClick={[Function]}
      >
        Download submitted response (Excel)
      </a>
    </li>
    <li>
      <a
        href="something/api/exportlead/1"
        onClick={[Function]}
      >
        Download all responses from your organisation (Excel)
      </a>
    </li>
  </ul>
  <h2>
    What happens next?
  </h2>
  <p>
    We will review all the submissions received for this consultation. Our response will be published on the website around the time the guidance is published.
  </p>
  <hr />
</Fragment>
`;

exports[`[ClientApp]  Submitted page should match snapshot for organisation lead when displayed on the review page without any responses from their org 1`] = `
<Fragment>
  <h2>
    Your response was submitted 
     on 
    <n
      ago={false}
      calendar={false}
      decimal={false}
      element={null}
      filter={[Function]}
      format="D MMMM YYYY"
      fromNow={false}
      interval={60000}
      local={false}
      onChange={[Function]}
      titleFormat=""
      toNow={false}
      trim={false}
      unit={null}
      unix={false}
      utc={false}
      withTitle={false}
    />
    .
  </h2>
  <ul
    className="list list--unstyled"
  >
    <li>
      <a
        href="something/api/exportexternal/1"
        onClick={[Function]}
      >
        Download submitted response (Excel)
      </a>
    </li>
  </ul>
  <h2>
    What happens next?
  </h2>
  <p>
    We will review all the submissions received for this consultation. Our response will be published on the website around the time the guidance is published.
  </p>
  <hr />
</Fragment>
`;

exports[`[ClientApp]  Submitted page should match snapshot for organisation lead when displayed on the submitted page 1`] = `
<Fragment>
  <h2>
    Your response was submitted 
     on 
    <n
      ago={false}
      calendar={false}
      decimal={false}
      element={null}
      filter={[Function]}
      format="D MMMM YYYY"
      fromNow={false}
      interval={60000}
      local={false}
      onChange={[Function]}
      titleFormat=""
      toNow={false}
      trim={false}
      unit={null}
      unix={false}
      utc={false}
      withTitle={false}
    />
    .
  </h2>
  <ul
    className="list list--unstyled"
  >
    <li>
      <Link
        data-qa-sel="review-submitted-comments"
        to="/1/review"
      >
        Review your response
      </Link>
    </li>
    <li>
      <a
        href="something/api/exportexternal/1"
        onClick={[Function]}
      >
        Download submitted response (Excel)
      </a>
    </li>
    <li>
      <a
        href="something/api/exportlead/1"
        onClick={[Function]}
      >
        Download all responses from your organisation (Excel)
      </a>
    </li>
  </ul>
  <h2>
    What happens next?
  </h2>
  <p>
    We will review all the submissions received for this consultation. Our response will be published on the website around the time the guidance is published.
  </p>
  <hr />
</Fragment>
`;
